### ISSUE TITLE
trident checksumupdate -> trident update

### ISSUE THREAD

**Issue by nevrome** [poseidon-framework/poseidon-hs#110]
I'm thinking a lot about trident when showering. So this is my latest shower thought:
The ongoing discussion regarding the [poseidon format](https://github.com/poseidon-framework/poseidon2-schema) will most likely cause more changes to available packages. For our own convenience I would like to make `trident checksumupdate` more powerful and rename it to `trident update`.

Here's the interface I have in mind:

```bash
trident update -d . \
  --poseidonVersion "1.2.3" \
  --version "major" (or "minor" or "patch") \
  --checksumUpdate \
  --date "2023-05-17"
```

This should make it more easy to update a lot of packages a once. Especially the version incrementing would be extremely useful. I believe this would be fairly easy to implement, but really useful in the future.

Maybe we should also automatically create a CHANGELOG file - not so sure about that. Could be really useful in the long run, but is a bit more work.

**Comment by nevrome** [comment]
I thought again today how extremely useful this would be to update packages. And we should ABSOLUTELY create and/or update the CHANGELOG file.

```
trident update -d . \
  --poseidonVersion "1.2.3" \ (default if nothing is selected: do nothing)
  --version "major" (or "minor" or "patch") \ (default: check if any checksum changed and if so update minor)
  --checksumUpdate \ (needs to be pared with a --ignoreGeno option)
  --date "2023-05-17" (default: today)
  --newContributor "Name,Email;Name,Email,..."
  --updateLog "I changed this and that: ..."
```

**Comment by stschiff** [comment]
Hmm, OK. Late to the game. Definitely looks like a slick and useful interface! I guess I would have thought that just manually tweaking the YML file would be around the same amount of keyboard strokes, but of course it's nice to have this process of updating packages available in a somewhat more controlled way.

**Comment by nevrome** [comment]
Hm - whenever we do somthing as simple as incrementing the `poseidonVersion`, we also have to increment the `packageVersion` and the `lastModified` date (just to make sure that fetch triggers a redownload for all of them). As every package has a different `packageVersion` and `lastModified` entry, that means individually changing 150+ files. So we're talking about a ton (!!) of keystrokes :stuck_out_tongue:

I would argue `update` is a very important key feature to keep our repository up-to-date.

**Comment by stschiff** [comment]
Ah OK... you mean for bulk editing all packages. OK, fair point. But why would we have to change the `packageVersion` every time we increment `poseidonVersion`? Many changes in the poseidon schema do not require changes in existing packages, do they?

**Comment by nevrome** [comment]
Ja - this function is mostly relevant for bulk updates. 

I think any change in a package warrants a version update. Package X in version Y should always be exactly the same, no matter where or when.

**Comment by stschiff** [comment]
Yes, I get it. Updating the Yaml file is itself a change in the package, so needs a package version update. One level more meta than I figured. Alright!

**Comment by nevrome** [comment]
Implemented in #111 and documented here: https://github.com/poseidon-framework/poseidon-framework.github.io/commit/22c6bfa79a704942319cd7560fc2f54c7c4dc8a4
